### PR TITLE
pad_left based on attention_mask

### PR DESCRIPTION
**Pull request by as-suvorov**

### CODE REVIEW

**Review by Wovchena — APPROVED** [review]

File: tests/python_tests/test_generate_api.py [hunk 1]
```diff
@@ -548,3 +548,32 @@ def test_unicode_pybind_decoding():
...
+        "diversity_penalty": 1.0,
+        # phi 1_5 has no eos_token_id in model configuration
+        # ov genai will detect eos_token_id from tokenizer config
+        # hf implementation doesn't fetch it from tokenizer config and defaults to None
+        # align ov genai and hf by setting eos_token_id explicitly
+        "eos_token_id": 50256,
+    }
+
+    models[2].pad_token = models[2].eos_token
+    run_hf_ov_genai_comparison_batched(models, config, prompts)
```
Comment (Wovchena) [inline tests/python_tests/test_generate_api.py:578]: I think the problem is that you have both OV and transformers models performing inference while `test_unicode_pybind_decoding()` uses ov only. You can align their configs to run both in greedy mode to verify that.

File: src/cpp/src/tokenizer.cpp [hunk 2]
```diff
@@ -249,6 +255,11 @@ class Tokenizer::TokenizerImpl {
             if (m_eos_token_id == -1 && content == m_eos_token)
                 m_eos_token_id = std::stoi(key);
         }
+
+        // if pad_token_id not found use eos_token_id as pad_token_id
```
Comment (ilya-lavrenov) [inline src/cpp/src/tokenizer.cpp:259]: please, create a ticket for OV tokenizers and insert TODO to update / drop this code once the ticket is fixed

File: tests/python_tests/test_generate_api.py (hunk 1 above)
Comment (as-suvorov) [reply tests/python_tests/test_generate_api.py:579]: I skipped test. I propose to merge this PR with skipped test. I'm working on enabling it in the next PR.

File: src/cpp/src/tokenizer.cpp (hunk 2 above)
Comment (as-suvorov) [reply src/cpp/src/tokenizer.cpp:201]: Ticket created

Comment (as-suvorov) [reply src/cpp/src/tokenizer.cpp:201]: CVS-144174

**Pull request merged**